clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -482,6 +497,47 @@ private void process(final GroupMetadataUpdateEvent event) {
                 requestManagersSupplier);
     }
 
+    // auxiliary interface for testing
+    interface ApplicationEventHandlerSupplier {
```

Comment:
It is nor clear to me why we need a supplier here. Could you elaborate?

Reply:
I want to inject a custom implementation of the applicationEventHandler. It needs to work for both prod and testing so that we can actually test the code we are running. The instantiation for prod depends on arguments that are created inside the constructor of AsyncKafkaConsumer. If I do not define this supplier, I cannot inject the constructor of the "real" ApplicationEventHandler for prod.